### PR TITLE
Profile

### PR DESCRIPTION
**Pull request by shansty**
Добавила app.get('/profile/:id', (req, res) => {} и app.patch('/profile/:id', (req, res) => {}

### CODE REVIEW

File: src/Auth.js [hunk 1]
```diff
@@ -2,70 +2,78 @@ const express = require('express');
...
+          return res.status(403).json({ message: `Invalid data + ${token}`});
+        }
+      let id = decoded.id;
+      let user = usersRegister.find(el => {
+        return el.id = id;
+      })
+      res.json({user})
+    });
+    } catch (err) {
+      console.error(err);
```
Comment (busha98) [inline src/Auth.js:53]: We need send response with error

File: src/Auth.js [hunk 2]
```diff
@@ -2,70 +2,78 @@ const express = require('express');
...
+
+
+
 app.post('/login', (req, res) => {
 
     const email = req.body.email;
     const password = req.body.password;
 
-    let user = usersLogin.find(el => {
+    let user = usersRegister.find(el => {
```
Comment (busha98) [inline src/Auth.js:65]: It's better to name this array as `users`

File: src/Auth.js [hunk 3]
```diff
@@ -2,70 +2,78 @@ const express = require('express');
...
+      let id = user.id;
+
     if (user) {
         // Generate a JWT token
-        tokenUser = jwt.sign( {email}, secretKey, { expiresIn: '24h' });
+        token = jwt.sign( {id}, secretKey, { expiresIn: '24h' });
     
         // Return the token to the client
-        res.json({ tokenUser });
+        res.json({ token: token });
```
Comment (busha98) [inline src/Auth.js:76]:
```suggestion
        res.json({ token });
```

File: src/Auth.js [hunk 4]
```diff
@@ -75,21 +83,9 @@ app.post('/login', (req, res) => {
 // Регистрация, если пользователя с таким именем нету
 app.post('/register', (req, res) => {
 
-
-    //и это не працуе, куда надо пихать этот ебучий валидатор
```
Comment (busha98) [inline src/Auth.js:79]: ))))))))

File: src/Auth.js [hunk 5]
```diff
@@ -108,14 +104,31 @@ app.post('/register', (req, res) => {
...
   }
 });
 
+
+app.patch('/profile/:id', (req, res) => {
+
+    const params = req.body;
+    const id =  req.params.id;
+
+    let user = usersRegister.find(el => {
```
Comment (busha98) [inline src/Auth.js:121]:
```
    let ind = usersRegister.findIndex(el => {
      return el.id === id;
    })
   ```

File: src/Auth.js [hunk 6]
```diff
@@ -108,14 +104,31 @@ app.post('/register', (req, res) => {
...
     usersRegister.push(newUser);
     console.log(usersRegister)
-    res.json({message: `${fullName}, Вы успешно зарегистрировались!`})
+    res.json({message: `${userName}, Вы успешно зарегистрировались!`})
   }
 });
 
+
+app.patch('/profile/:id', (req, res) => {
+
```
Comment (busha98) [inline src/Auth.js:117]: add validation + authorization token.

user can call PATCH profile only for his own Id

File: src/Auth.js (hunk 3 above)
Comment (shansty) [reply src/Auth.js:76]: fixed

File: src/Auth.js (hunk 6 above)
Comment (shansty) [reply src/Auth.js:117]: fixed

File: src/Auth.js [hunk 7]
```diff
@@ -108,14 +82,67 @@ app.post('/register', (req, res) => {
...
+app.get('/profile/:id', (req, res) => {
+  const authHeader = req.headers['authorization'];
+  const token = authHeader && authHeader.split(' ')[1];
+  const id = req.params.id;
+  try {
+    jwt.verify(token, secretKey, (err, decoded) => {
+      if (err) {
+        return res.status(403).json({ message: `Invalid data + ${token}`});
+      } else {
+      if (id == decoded.id) {
```
Comment (busha98) [inline src/Auth.js:103]: what if else ?

File: src/Auth.js [hunk 8]
```diff
@@ -108,14 +82,67 @@ app.post('/register', (req, res) => {
...
+        let user = users.find(el => {
+          return el.id == id;
+        })
+        res.json({user})
+      }}
+    });
+  } catch (err) {
+    res.json("Error")
```
Comment (busha98) [inline src/Auth.js:111]: there shouldn't be status code 200

File: src/Auth.js [hunk 9]
```diff
@@ -108,14 +82,67 @@ app.post('/register', (req, res) => {
...
+            return el.id == id;
+          })
+          console.log(params)
+          users[ind] = {...users[ind], ...params}
+          // res.json({message: `Данные пользователя обновлены`})
+          res.json(users[ind])
+        }
+      });
+    } catch (err) {
+      console.error(err);
```
Comment (busha98) [inline src/Auth.js:142]: res.error

File: src/Auth.js [hunk 10]
```diff
@@ -108,14 +82,67 @@ app.post('/register', (req, res) => {
...
+
+    console.log(params)
+    console.log(id)
+
+    try {
+      jwt.verify(token, secretKey, (err, decoded) => {
+        if (err) {
+          return res.status(403).json({ message: `Invalid data + ${token}`});
+        }
+        if (id == decoded.id) {
```
Comment (busha98) [inline src/Auth.js:131]: else there no reponse (we need it)

File: src/Auth.js [hunk 11]
```diff
@@ -108,14 +82,71 @@ app.post('/register', (req, res) => {
...
+    const params = req.body;
+    const id =  req.params.id;
+
+    console.log(params)
+    console.log(id)
+
+    try {
+      jwt.verify(token, secretKey, (err, decoded) => {
+        if (err) {
+          return res.status(403).json({ message: `Invalid data`});
```
Comment (busha98) [inline src/Auth.js:131]:
```suggestion
          res.status(403).json({ message: `Invalid data`});
          return;
```

File: src/Auth.js [hunk 12]
```diff
@@ -108,14 +82,71 @@ app.post('/register', (req, res) => {
...
 
+
+app.get('/profile/:id', (req, res) => {
+  const authHeader = req.headers['authorization'];
+  const token = authHeader && authHeader.split(' ')[1];
+  const id = req.params.id;
+  try {
+    jwt.verify(token, secretKey, (err, decoded) => {
+      if (err) {
+        return res.status(403).json({ message: err.message});
```
Comment (busha98) [inline src/Auth.js:101]:
```suggestion
        res.status(403).json({ message: err.message});
        return;
```

**Review by busha98 — APPROVED** [review]

**Pull request merged**